java-checks-test-sources/src/main/java/checks/spring/ModelAttributeNamingConventionForSpELCheck.java

```diff
@@ -0,0 +1,47 @@
+package checks.spring;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+class ModelAttributeNamingConventionForSpELCheck {
+
+  private static final String MY_LEGAL_CONSTANT = "legalName";
+  private static final String MY_ILLEGAL_CONSTANT = "a-b";
+
+  public void foo(org.springframework.ui.Model model) {
+
+    model.addAttribute(File.separator, 42); // Compliant - can not resolve
+    model.addAttribute(MY_LEGAL_CONSTANT, 0); // Compliant
+    model.addAttribute(MY_ILLEGAL_CONSTANT, 0); // Noncompliant
+
+    model.addAllAttributes(Map.of("m", 42, "a", 22)); // Compliant
+    model.addAllAttributes(getMap()); // Compliant
+
+    model.addAllAttributes(Map.of(" m", 42, " a", 22)); // Noncompliant [[sc=28;ec=54]] {{Attribute names must begin with a letter (a-z, A-Z), underscore (_), or dollar sign ($)
+                                                        // and can be followed by letters, digits, underscores, or dollar signs.}}
+
```

Comment:
To go further into the stupidly complicated what about supporting:
```suggestion

model.addAllAttributes(Map.of(MY_ILLEGAL_CONSTANT, 42)); // Noncompliant
```